### ISSUE TITLE
Theme-config for have a possible to make other themes

### ISSUE THREAD

**Issue by blackcrack** [mwalbeck/nextcloud-breeze-dark#141]
Hi,

i have take you're offizial theme and use it for my theme where i have use it on /theme folder.
now it is so, it need a "defaults.php" where we can setting up different things like the 
set at the loginpage at bottom .. and now have i see it, is more in the structure bring it in the name 
"Breeze Dark" and "breezedark" as example 
in 
 [themplates admin](https://github.com/mwalbeck/nextcloud-breeze-dark/blob/master/templates/admin.php) and [themplates personal](https://github.com/mwalbeck/nextcloud-breeze-dark/blob/master/templates/personal.php)

At this step could it be possible to create a defaults.php
with variables to have the possible for adjust the whole theme 
Name:"Breete Dark" 
ID : "breezedark"
in the defaults.php
in this form
```
/**
         * Returns the base URL
         * @return string URL
         */
        public function getBaseUrl() {
                return 'https://nextcloud.com';
        }
```
and with
```
/**
         * Returns the base Name
         * @return string Theme Name
         */
        public function themename() {
                return 'Breeze Dark';
        }
/**
         * Returns the base Theme ID
         * @return string Theme id
         */
        public function getBasethemeid() {
                return 'breezedark';
        }
```

this funtions could you include in the app.php then could you 
change it to
```
include(../defaults.php)
Util::addStyle('getBasethemeid()', 'guest');
Util::addStyle('getBasethemeid()', 'server');
``` 

For have a theme-exemple with adjust possibility's for making other themes ?
 include the whole informations from the ./themes/example/defaults.php

therewith could you be the Mother of  "app-theming" if you be so good for making this :)

i have alreaddy started to porting my blackysgate NeXtrealTime to apps/nextrealtime

with this issue makes you the other Designer more easy to create a own theme with you
Mother-Theme \*nod*  if you update this  and add a defaults.php in the root-dir of you're theme folder.

best regards
Blacky

**Comment by mwalbeck** [comment]
I have a few points on this.

1. The defaults.php I think you are thinking about is specific to the theming folder and isn't used in the apps.

2. In the nc18 and nc19 version it is pretty simple already to use the theme as a base for another theme. It's really just a question of search and replace. There are 2 places I'll have a look where I can probably use a variable instead of hardcoded value, but a search and replace will still be needed.

3. The example you're providing is only applicable to the nc17 version of the theme, which I no longer support. The nc17 codebase is quite different from the nc18 and nc19 codebases.

In summary, it's already quite easy to use the theme as a base for another theme. It'll never be as simple as changing a single variable and including an extra file for containing said variable(s) is going to complicate things more than necessary.

**Comment by blackcrack** [comment]
1. right.

2. i have allready modify you're theme :) ... but in NC17 :\

3. ah okey, did now know..

and the 2nd summary, i use up to NC18 my old theme in the /Theme folder..
and 2nd the close the issue ! :) :+1: 

best regards
Blacky
